### PR TITLE
Move CLI code to cmd/internal and move application code to internal/app

### PR DESCRIPTION
**Pull request by bauerm97**
**Description of the Pull Request (PR):**

In my ongoing quest to make Singularity code cleaner and a bit more Go idiomatic, I've decided that all code containing the `cobra.Command` structs should live within `cmd/internal/cli`. The reason for that is because we actually have several main packages which need access to those structs:

1. `etc/bash_completion/bash_completion.go`: This file is the main package which uses cobra's bash completion. Since this is actually a main package, we want to put this in the canonical location of `cmd`. Therefore, I've moved it to `cmd/bash_completion/bash_completion.go`.
2. `docs/makeDocs/makeDocs.go`: This file is similar in function to `bash_completion.go`, however it generates man pages instead. I've similarly moved this to `cmd/man_pages/man_pages.go`
3. `cmd/singularity/..`: singularity main cli binary

This PR moves all the CLI code which was previously in `internal/app/singularity` -> `cmd/internal/cli`. Then, I take a first pass at decoupling specific application code from CLI code by moving application code for the `singularity capability ...` command group into `internal/app/singularity`.

@mikegray Can you please take a look at this and make sure I'm not breaking macOS building since I've moved a few files around and added a new one.

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](../CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](../CHANGELOG.md) if necessary according to the [Contribution Guidelines](../CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](../CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](../CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

**Review by mikegray — COMMENTED** [review]
I pulled the PR locally and it does build successfully on macOS.

File: cmd/internal/cli/capability.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
```
Comment (ArangoGutierrez) [inline cmd/internal/cli/capability.go:1]: Update Copyright -> 2019

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]
small tip

File: cmd/internal/cli/capability_add.go [hunk 2]
```diff
@@ -36,7 +39,18 @@ var CapabilityAddCmd = &cobra.Command{
...
+			User:  CapUser,
+			Group: CapGroup,
+			Desc:  CapDesc,
+		}
+
+		if err := singularity.CapabilityAdd(buildcfg.CAPABILITY_FILE, c); err != nil {
+			sylog.Fatalf("Unable to add capabilities: %s", err)
+		}
+
+		// manageCap(args[0], capAdd)
```
Comment (ikaneshiro) [inline cmd/internal/cli/capability_add.go:53]: I assume this commented code is no longer needed

File: cmd/internal/cli/capability_drop.go [hunk 3]
```diff
@@ -33,10 +36,21 @@ func init() {
...
+			User:  CapUser,
+			Group: CapGroup,
+			Desc:  CapDesc,
+		}
+
+		if err := singularity.CapabilityDrop(buildcfg.CAPABILITY_FILE, c); err != nil {
+			sylog.Fatalf("Unable to drop capabilities: %s", err)
+		}
+
+		// manageCap(args[0], capDrop)
```
Comment (ikaneshiro) [inline cmd/internal/cli/capability_drop.go:53]: Same as above

**Review by ikaneshiro — APPROVED** [review]
Just a couple nit picks, otherwise LGTM

**Comment by tri-adam** [comment]
@bauerm97, LGTM, can you `merge`/`rebase` to resolve the conflicts?

**Comment by bauerm97** [comment]
Added a commit updating `pkg/util/capabilities` a bit based on feedback from @tri-adam . New tests have 95% LoC coverage.

File: pkg/util/capabilities/config.go [hunk 4]
```diff
@@ -0,0 +1,211 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
```
Comment (ArangoGutierrez) [inline pkg/util/capabilities/config.go:1]: 2019

**Comment by bauerm97** [comment]
Closed and replaced by #2618

**Pull request closed without merging**